django/db/models/query.py

```diff
@@ -818,12 +818,33 @@ def _combinator_query(self, combinator, *other_qs, all=False):
         return clone
 
     def union(self, *other_qs, all=False):
+        # If the query is an EmptyQuerySet, replace the query with first
```

Comment:
@timgraham  My second idea is to start from the first nonempty queryset. What do you think?